### PR TITLE
[Clang importer] Infer @_downgrade_exhaustivity_check for new-to-2017 enum elements in Swift 3

### PR DESCRIPTION
**Pull request by DougGregor**
In Swift 3 mode, infer @_downgrade_exhaustivity_check for any enum
elements that were introduced in the 2017 SDKs (macOS 10.13, iOS 11.0,
tvOS 11.0, watchOS 4.0). This is a stop-gap until we get "open" enums,
but serves an important source-compatibility use case.

Fixes rdar://problem/32824207.

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please smoke test

File: test/ClangImporter/availability_open_enums_macos_swift3.swift [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// RUN: %target-swift-frontend(mock-sdk: %clang-importer-sdk) -typecheck -verify -I %S/Inputs/custom-modules -swift-version 3 -target x86_64-apple-macosx10.13 %s
```
Comment (jrose-apple) [inline test/ClangImporter/availability_open_enums_macos_swift3.swift]: Any way to test all the Apple platforms? It seems like `@available` could avoid having to hardcode a target triple.

Comment (DougGregor) [inline test/ClangImporter/availability_open_enums_macos_swift3.swift]: Hmm... that's a good point. I think I can drop the specific target in the RUN line and fuzzy-match the output to get it to run on all platforms. That would make me much more comfortable.

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

File: test/ClangImporter/availability_open_enums_macos_swift3.swift (hunk 1 above)
Comment (DougGregor) [inline test/ClangImporter/availability_open_enums_macos_swift3.swift]: Yep, it was trivial. Thank you!

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

**Review by jrose-apple — APPROVED** [review]

**Pull request merged**